### PR TITLE
docs: update developer guide to include implementation for the export feature

### PR DESCRIPTION
**Pull request by danielleloh**
* add implementation details for export command
* move export sequence diagram down to aid in the explanation of the implementation details
* command exception object was unused in the export sequence. so, remove the command exception class.

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (Zer0Legion) [inline docs/DeveloperGuide.md:1]: insightful

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -188,6 +170,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+The `excute` method calls the `Model#getFilteredPersonList()` to retrieve the `filteredPersons` list stored in `Model`. This lists stores the relevant persons currently displayed in the `PersonListPanel`.
+It then creates a temporary `AddressBook` object and iterates through the `filteredPerons` list to add each person from the list into the AddressBook.
+
+#### JSON Serialization
+
+The information in the temporary `AddressBook` is written into a JSON file, `filteredaddressbook.json` using the `JsonAddressBookStorage#writeToJsonFile()` method.
+
+#### CSV Conversion
+
+The `export` feature relies on the Jackson Dataformat CSV module and the Jackson Databind module to read the JSON file and write the information into a CSV file.
```
Comment (Pughal77) [inline docs/DeveloperGuide.md:202]: Good including our dependencies in the DG

**Review by Pughal77 — COMMENTED** [review]
Nice detailed description of export implementation

**Review by delishad21 — COMMENTED** [review]
Well done, detailed and easy to understand

**Pull request merged**